lib/date_time/date_time_core.c

```diff
@@ -44,16 +44,14 @@ static void date_time_core_notify_event(enum date_time_evt_type time_source)
 
 static void date_time_core_schedule_update(void)
 {
-	/* If periodic updates are requested in the configuration
-	 * and next update hasn't been already scheduled.
+	/* (Re-)schedule time update work
+	 * if periodic updates are requested in the configuration.
 	 */
-	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0 &&
-	    !k_work_delayable_is_pending(&time_work)) {
-
-		LOG_DBG("New date time update in: %d seconds",
+	if (CONFIG_DATE_TIME_UPDATE_INTERVAL_SECONDS > 0) {
```

Comment:
This function is called from several different places and depending on whether the timer should be **re**scheduled, a `k_work_delayable_is_pending()` check is sometimes needed before the call. What do you think about doing the `k_work_delayable_is_pending()` check inside this function, but making it optional by adding a parameter for it, for example:

`static void date_time_core_schedule_update(bool reschedule)`

I just think that it could be clearer to have the `k_work_delayable_is_pending()` check inside `date_time_core_schedule_update()`.

Reply:
Yep, that would be a little clearer. I'll do the changes.

Reply:
I added a `check_pending` boolean there and if it's true and the work is already pending, we will exit from that function.